lib/Optimizer/GraphOptimizer/GraphOptimizer.cpp

```diff
@@ -409,6 +409,41 @@ bool SinkCode::run(Function *F, const CompilationContext &cctx) {
   // For each node:
   for (auto &N : nodes) {
     auto *node = &N;
+    // Sink Reshape below BatchNormalization.
+    if (auto *BN = dyn_cast<BatchNormalizationNode>(node)) {
+      if (auto *RS = dyn_cast<ReshapeNode>(BN->getInput())) {
+        auto inDims = RS->getInput().dims();
+        auto outDims = RS->getType(0)->dims();
+        // Reshape should not change the Channel Idx dimensions.
+        // Num input channels is equal to num output channels.
+        // BatchNorm ChannelIdx is C in NHWC or NWC format.
```

Comment:
We also use NCHW, so we don't know for sure it will always be the final dimension. Perhaps we could just check that dim `BN->getChannelIdx()` is the same before and after, and that there are the same number of dimensions before and after?

Reply:
"check that dim BN->getChannelIdx() is the same before and after" - Makes sense.
"same number of dimensions before and after" - may not be the case always. Reshape could flatten HW dimensions to W/H, and it does not affect the optimization.

Reply:
Ok sure, but we still should be able to modify this to handle NCHW too, right? We can sink the reshape if C stays the same. So for NHWC/NWC we can always check the final dimension like you are. For NCHW/NCW we can always check the second dimension.

Reply:
Ahh... understood. Done.

Reply:
@jfix71 , Were you able to take a look at it?